src/battle_script_commands.c

```diff
@@ -8426,6 +8427,57 @@ static void Cmd_various(void)
             gBattlescriptCurrInstr += 7;    // exit if loop failed (failsafe)
         }
         return;
+    case VARIOUS_PHOTON_GEYSER_CHECK:
+    {
+        u32 attStat = gBattleMons[gActiveBattler].attack;
+        u8 atkStage = gBattleMons[gActiveBattler].statStages[STAT_ATK];
+        u32 spaStat = gBattleMons[gActiveBattler].spAttack;
+
+        attStat *= gStatStageRatios[atkStage][0];
+        attStat /= gStatStageRatios[atkStage][1];
+
+        atkStage = gBattleMons[gActiveBattler].statStages[STAT_SPATK];
+        spaStat *= gStatStageRatios[atkStage][0];
+        spaStat /= gStatStageRatios[atkStage][1];
+
+        if (attStat > spaStat)
+            gSwapDamageCategory = TRUE;
```

Comment:
Is this variable needed? Can't this be moved to damage calculator?

Reply:
I don't think so, the move is calculated without any other stat modifiers added in. I was thinking I could combine the two various commands with an if statement to continue separately if it was SSA.

Reply:
If this is moved to the damage calculator, checking for the physical / special split outside of calculating damage does not happen. This is mostly specific to SSA making contact (`IsMoveMakingContact`), which is why we are using a global variable to address the change in split. The global variable we use also modifies `GetBattleMoveSplit`, which the damage calculator then uses to choose the correct split for the attack.
Perhaps I am not aware of when calculating damage takes places relative to making contact and such? But implementing this in those places makes sense to me.  I am still thinking over combine the various cases to be more general, with an `if` statement to address SSA specifically.